### PR TITLE
JAMES-2167 MailQueue should preserve null sender

### PR DESCRIPTION
**Pull request by chibenwa**

### CODE REVIEW

**Comment by chibenwa** [comment]
```
[3f58718ddbd5afa66408a7c5210e27afb0e8564d] [ERROR] Errors: 
[3f58718ddbd5afa66408a7c5210e27afb0e8564d] [ERROR]   DomainUserMaildirMailboxManagerTest>MailboxManagerTest.creatingConcurrentlyMailboxesWithSameParentShouldNotFail:988 » Execution
[3f58718ddbd5afa66408a7c5210e27afb0e8564d] [ERROR]   FullUserMaildirMailboxManagerTest>MailboxManagerTest.creatingConcurrentlyMailboxesWithSameParentShouldNotFail:988 » Execution
```

Hmm test this please

File: core/src/main/java/org/apache/james/core/MailAddress.java [hunk 1]
```diff
@@ -356,6 +358,9 @@ public boolean equals(Object obj) {
             return toString().equalsIgnoreCase(theString);
         } else if (obj instanceof MailAddress) {
             MailAddress addr = (MailAddress) obj;
+            if (isNullSender() && addr.isNullSender()) {
```
Comment (mbaechler) [inline core/src/main/java/org/apache/james/core/MailAddress.java]: you are missing a unit test for that (and maybe for the whole equals method)

File: server/queue/queue-jms/src/main/java/org/apache/james/queue/jms/JMSMailQueue.java [hunk 2]
```diff
@@ -412,22 +412,25 @@ protected void populateMail(Message message, MailImpl mail) throws JMSException
...
-
+        mail.setSender(getMailSender(message.getStringProperty(JAMES_MAIL_SENDER), mail));
         mail.setState(message.getStringProperty(JAMES_MAIL_STATE));
+    }
 
+    private MailAddress getMailSender(String sender, Mail mail) {
+        if (sender == null || sender.trim().length() <= 0) {
+            return null;
+        }
+        if (sender.equals(MailAddress.NULL_SENDER_AS_STRING)) {
```
Comment (mbaechler) [inline server/queue/queue-jms/src/main/java/org/apache/james/queue/jms/JMSMailQueue.java]: doesn't `MailAddress.nullSender().equals(sender)` return the same result ?

Comment (chibenwa) [reply server/queue/queue-jms/src/main/java/org/apache/james/queue/jms/JMSMailQueue.java]: No. Sender is a string here...

**Comment by chibenwa** [comment]
No more tests are being run here...

File: server/queue/queue-jms/src/main/java/org/apache/james/queue/jms/JMSMailQueue.java (hunk 2 above)
Comment (chibenwa) [reply server/queue/queue-jms/src/main/java/org/apache/james/queue/jms/JMSMailQueue.java]: Well, that's the entire goal of that method ^^

Comment (chibenwa) [reply server/queue/queue-jms/src/main/java/org/apache/james/queue/jms/JMSMailQueue.java]: `private MailAddress getMailSender(String sender, Mail mail)`

**Pull request closed without merging**